pkg/client/image/image.go

```diff
@@ -35,6 +41,40 @@ func (c *Client) Delete(id string) error {
 	return c.client.Delete(id, c.instanceID)
 }
 
+//func ImportImage imports image from S3 Instance
+func (c *Client) ImportImage(instanceID, imageName, s3Filename, region, accessKey, secretKey, bucketName, osType, diskType string) (*models.Image, error) {
+	var source = "url"
+	var body = models.CreateImage{
+		ImageName:     imageName,
+		ImageFilename: s3Filename,
+		Region:        region,
+		AccessKey:     accessKey,
+		SecretKey:     secretKey,
+		BucketName:    bucketName,
+		OsType:        osType,
+		DiskType:      diskType,
+		Source:        &source,
+	}
+
+	params := p_cloud_images.NewPcloudCloudinstancesImagesPostParamsWithTimeout(pkg.TIMEOUT).WithCloudInstanceID(instanceID).WithBody(&body)
+	resp1, resp2, err := c.session.Power.PCloudImages.PcloudCloudinstancesImagesPost(params, ibmpisession.NewAuth(c.session, instanceID))
+
+	if err != nil {
+		return &models.Image{}, err
+	}
+
+	if resp1 != nil {
+		klog.Errorf("Failed to intiate the import job")
+		return &models.Image{}, errors.New("Failed to initiate the import job")
```

Comment:
Returning nil is good enough I feel: `return nil, err`

Reply:
done